pkg/volume/util/util.go

```diff
@@ -96,29 +97,44 @@ func UnmountPath(mountPath string, mounter mount.Interface) error {
 // IsNotMountPoint will be called instead of IsLikelyNotMountPoint.
 // IsNotMountPoint is more expensive but properly handles bind mounts.
 func UnmountMountPoint(mountPath string, mounter mount.Interface, extensiveMountPointCheck bool) error {
-	if pathExists, pathErr := PathExists(mountPath); pathErr != nil {
+	pathExists, pathErr := PathExists(mountPath)
+	corruptedMnt := isCorruptedMnt(pathErr)
+	return UnmountMountPath(mountPath, mounter, extensiveMountPointCheck, pathExists, pathErr, corruptedMnt)
+}
+
+// UnmountMountPoint is a common unmount routine that unmounts the given path and
+// deletes the remaining directory if successful.
+// if extensiveMountPointCheck is true
+// IsNotMountPoint will be called instead of IsLikelyNotMountPoint.
+// IsNotMountPoint is more expensive but properly handles bind mounts.
+// pathExists should be true if the mountPath existed,else false
+// pathErr is the error when stats the mountPath
+// corruptedMnt should be true if the mountPath is a corrupted mount point
+func UnmountMountPath(mountPath string, mounter mount.Interface, extensiveMountPointCheck bool, pathExists bool, pathErr error, corruptedMnt bool) error {
```

Comment:
Does this need to be a public method? I don't see any external callers.

Reply:
Also do you need to have corruptedMnt as an argument since you can get it from pathErr?

Reply:
thanks for your review!
I have changed it to private.
I have corruptedMnt as an argument  for make the unit test easily.

Reply:
in your unit test, you are passing in pathErr already.  It seems simpler to just derive corruptedMnt from pathErr.  Then, you will also be testing `isCorruptedErr`.

Reply:
Oh, you are right! thank you!
I have refactored it 
@msau42 PTAL